src/sfizz/FilePool.h

```diff
@@ -346,40 +417,73 @@ class FilePool {
 
     // Signals
     volatile bool dispatchFlag { true };
-    volatile bool garbageFlag { true };
-    RTSemaphore dispatchBarrier;
-    RTSemaphore semGarbageBarrier;
+    RTSemaphore dispatchBarrier { 0 };
 
     // Structures for the background loaders
     struct QueuedFileData
     {
         QueuedFileData() noexcept {}
-        QueuedFileData(std::weak_ptr<FileId> id, FileData* data) noexcept
+        QueuedFileData(std::weak_ptr<FileId> id, const std::shared_ptr<FileData>& data) noexcept
         : id(id), data(data) {}
         std::weak_ptr<FileId> id;
-        FileData* data { nullptr };
+        std::shared_ptr<FileData> data { nullptr };
     };
 
     using FileQueue = atomic_queue::AtomicQueue2<QueuedFileData, config::maxVoices>;
     aligned_unique_ptr<FileQueue> filesToLoad;
 
     void dispatchingJob() noexcept;
-    void garbageJob() noexcept;
     void loadingJob(const QueuedFileData& data) noexcept;
     std::mutex loadingJobsMutex;
     std::vector<std::future<void>> loadingJobs;
     std::thread dispatchThread { &FilePool::dispatchingJob, this };
-    std::thread garbageThread { &FilePool::garbageJob, this };
 
-    SpinMutex garbageAndLastUsedMutex;
-    std::vector<FileId> lastUsedFiles;
-    std::vector<FileAudioBuffer> garbageToCollect;
+public:
+    struct GlobalObject
```

Comment:
I'm not the biggest fan of this name, it feel a bit vague. I guess that this is somehow an extension of the "global threadpool" which includes both the threadpool and the filepool, but maybe we can find another more precise name (or even just change the name of the per-instance "filepools" to something else).

As I'm reading more I might have other ideas :)

Reply:
I agree. but I am not good at English. Could you suggest one or more?

Reply:
I will look at the rest. My intuition is that "Filepool" can actually be the Global Object and that we have other names for the filepool reference in each synth.